### ISSUE TITLE
Review cursor commands may fail in Microsoft Edge under particular circumstances

### ISSUE THREAD

**Issue by k-kolev1985** [nvaccess/nvda#9066]
### Steps to reproduce:
1. Install the latest NVDA alpha snapshot.
2. Enable mouse tracking in NVDA's settings -> Mouse.
3. In NVDA's settings, under "Review Cursor", enable "Follow mouse cursor".
4. Open an HTML document in Microsoft Edge.
5. Use the mouse cursor to read a line (e.g. - the first one) from a multi-line paragraph of text.
6. Use the review cursor commands to read the previous and next line, word and character, respectively.

### Actual behavior:
NVDA plays its error sound and does not read the respective text. I'm attaching a log file with the error, with logging level set to "Debug" - hope it helps.

### Expected behavior:
NVDA reads the respective text in accordance with the respective review cursor command.

### System configuration
- Operating system: Windows 10 Pro version 1809 (build 17763.194), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".
- NVDA version: alpha-16418,75778041.
- Microsoft Edge version: 44.17763.1.0; Microsoft EdgeHTML 18.17763.
- Processor: Intel Core i5-4670 at 3.40GHz.
- RAM Memory: 8.00GB.
- Graphics: MSI GeForce GTX-1050TI Gaming X 4G, 4096MB dedicated memory + 4065MB shared system memory, desktop resolution set to 1920x1080.
- Sound Card: Realtek ALC887 at Intel Lynx Point PCH - High Definition Audio Controller.

#### Does the issue still occur after restarting your PC?
Yes, it does. And it also does with add-ons disabled.

#### Have you tried any other versions of NVDA?
Yes, the issue started occurring recently. It is probably a regression introduced by issue #8572.
[NVDA_Log_2018-12-13.log](https://github.com/nvaccess/nvda/files/2677313/NVDA_Log_2018-12-13.log)

**Comment by leonardder** [comment]
Did this work 100% properly before? In Edge, NVDA wasn't able to find the position of the mouse within a chunk of text.

**Comment by leonardder** [comment]
While this might have become more prevalent since #8572, I don't think it can be classified as a regression, i.e. it is likely that this is reproducible without the mouse somehow. A problem is that I'm not very well known with the current Edge code. May be @michaeldcurran can spread some light on this?

**Comment by k-kolev1985** [comment]
Well, you're probably right that it is not a regression from #8572. But I've tested with a portable copy of NVDA 2018.3.2 and after I move the mouse over a paragraph of text, I can then read in it via words and characters with the review cursor. Reading line by line is a bit trickier to make it work, but still possible, if I first move the review cursor to the next object and back to the current object/paragraph of text. But at least reading by words and characters worked before and now it does not.

**Comment by Adriani90** [comment]
Update with NVDA alpha-20015,56a62097 on Edge scromium stable 81:
When moving the mouse and then trying to navigate character by character or word by word with review commands, it works as expected. However, line by line does not work. Actually it seems document review disappears and we have just screen review or object review remaining. Pressing arrow keys in browse mode makes the document review to appear again and you can check that with nvda+nu lock 1 or nvda+num lock 7.
I guess this is not expected.

In Firefox for example, when moving the mouse the review cursor remains in document review mode but behaves as if it was in object review mode. You can check that also with nvda+numlock 1 and nvda+numlock 7. However, this is also not expected behavior.
cc: @jcsteh at least for the Firefox part.

**Comment by Adriani90** [comment]
In Crome 81 the document review mode also disappears when moving the mouse, same like in Edge cromium.

**Comment by Adriani90** [comment]
Now its working fine with NVDA last alpha in Windows 1 24 H2, probably due to #18271. I cannot reproduce anymore.
The only issue is
1. Move the mouse over the webpage
2. Directly after that, change between review modes with nvda+page up and down (laptop keyboard layout).

Actual: only screen review and object review are available, document review disappears.

Expected: document review should be available as well, since NVDA is in browse mode while moving the mouse. This is how it works in Firefox at least.

@jcsteh is this something Chrome needs to fix? Or can NVDA do something about it?

**Comment by jcsteh** [comment]
It's unclear what the expected behaviour is here. Is the expectation that the review cursor gets routed to the object under the mouse within document review? Or is the expectation that the review cursor can review the object under the mouse with object review? I'm guessing most users want the former: document review. But currently, mouse tracking doesn't use browse mode at all, so that would be inconsistent and probably explains why this behaves as it does now.

**Comment by Adriani90** [comment]
@jcsteh as far as I understand, in Firefox at least, when I move the mouse, document review is automatically active and using review cursor shows that I am in document review. This is so far consistent between browsers and I don't see a valid reason why this should be changed.

When switching between review modes after moving the mouse, in Firefox all three modes are still available (screen review, document review and object review). In Chrome and Edge however, document review disappears. I guess this shouldn't happen.

**Comment by jcsteh** [comment]
Ah. That might be because beTransparentToMouse isn't set for text leaf nodes in NVDA's Chromium code, but I'm not certain of that.
